usecases/configuration_test.go

```diff
@@ -0,0 +1,241 @@
+		Url:  "https://leadingregistrar",
+	}
+	test := components.CoreSystem{
+		Name: "test",
+		Url:  "https://test",
+	}
+
+	defaultConfig.CCoreS = append(defaultConfig.CCoreS, leadingRegistrar)
+	defaultConfig.CCoreS = append(defaultConfig.CCoreS, test)
+	defaultConfig.CName = sys.Name
+	defaultConfig.Protocols = sys.Husk.ProtoPort
+	os.Remove("systemconfig.json")
+	defaultConfigFile, err := os.Create("systemconfig.json")
+	if err != nil {
+		log.Fatalf("Encountered error while creating default config file")
+	}
+	defer defaultConfigFile.Close()
+
+	enc := json.NewEncoder(defaultConfigFile) // Create an encoder that allows writing to a file
+	enc.SetIndent("", "     ")                // Set proper indentation
+	err = enc.Encode(defaultConfig)           // Write defaultConfig template to file
+	if err != nil {
+		log.Fatalf("jsonEncode: %v", err)
+	}
+}
+
+type configureParams struct {
+	testCase        string
+	brokenSystem    bool
+	assetHasTraits  bool
+	assetNumber     int
+	createNewConfig bool
+	allowConfigRead bool
+	expectError     bool
+}
+
+func TestConfigure(t *testing.T) {
+	testParams := []configureParams{
+		// {testCase, brokenSystem, assetHasTraits, assetNumber, createNewConfig, allowConfigRead, expectError}
+		{"Best case, no errors", false, false, 1, true, true, false},
+		{"Missing asset", false, false, 0, false, true, true},
+		{"Good case, asset has traits", false, true, 0, true, true, false},
+		{"Config missing", false, false, 1, false, true, true},
+		{"Config missing, cant open or create", false, false, 1, false, false, true},
+		{"No Assets in config", false, false, 0, true, true, false},
+		{"Multiple Assets in config", false, false, 3, true, true, false},
+	}
+	defer os.Remove("systemconfig.json")
+	for _, testCase := range testParams {
+		testSys := createTestSystem(false)
+		if testCase.testCase == "Missing asset" {
+			testSys.UAssets = nil
+		}
+		if testCase.createNewConfig == true {
+			createConfig(&testSys, testCase.assetHasTraits, testCase.assetNumber)
+		}
+		if testCase.allowConfigRead == false {
+			os.OpenFile("systemconfig.json", os.O_RDWR|os.O_CREATE, 0000)
+		}
+		_, err := Configure(&testSys)
+		if testCase.expectError == false {
+			if err != nil {
+				t.Errorf("Expected no errors in best case, got: %#v", err)
+			}
+		} else {
+			if err == nil {
+				t.Errorf("Expected errors in testcase '%s' got none", testCase.testCase)
+			}
+		}
+		if (testCase.createNewConfig == true) || (testCase.allowConfigRead == false) {
+			err = os.Remove("systemconfig.json")
+			if err != nil {
+				t.Fatalf("failed while removing file")
+			}
+		}
+		if errors.Is(err, ErrNewConfig) {
+			err = os.Remove("systemconfig.json")
+			if err != nil {
+				t.Fatalf("failed while removing file")
+			}
+		}
+	}
+}
+
+func TestGetServiceList(t *testing.T) {
+	// getServicesList(uat components.UnitAsset) []components.Service
+	testSys := createTestSystem(false)
+	ua := (*testSys.UAssets["testUnitAsset"])
```

Comment:
Rather than calling that huge monster `createTestSystem`, I think you should just try create the single unitasset directly. If possible.